pandas/core/dtypes/dtypes.py

```diff
@@ -811,6 +811,51 @@ def __eq__(self, other: Any) -> bool:
             and tz_compare(self.tz, other.tz)
         )
 
+    def __from_arrow__(
+        self, array: pyarrow.Array | pyarrow.ChunkedArray
+    ) -> DatetimeArray:
+        """
+        Construct DatetimeArray from pyarrow Array/ChunkedArray.
+        """
+        import pyarrow
+        import pyarrow.types
```

Comment:
Nit: `is_timestamp` is accessible from the first `import pyarrow` as `pyarrow.types.is_timestamp`